src/core/connection/Generation.cpp

```diff
@@ -344,6 +344,41 @@ namespace Qv2ray::core::connection
                     inboundsList.append(socksInBoundObject);
                 }
 
+                // TPROXY
+                if (GlobalConfig.inboundConfig.useTPROXY)
+                {
+                   INBOUND  tproxyInBoundObject;
+                   tproxyInBoundObject.insert("listen", GlobalConfig.inboundConfig.tproxy_ip);
+                   tproxyInBoundObject.insert("port", GlobalConfig.inboundConfig.tproxy_port);
+                   tproxyInBoundObject.insert("protocol", "dokodemo-door");
+                   QString tproxy_network;
+                   if (GlobalConfig.inboundConfig.tproxy_use_tcp&&GlobalConfig.inboundConfig.tproxy_use_udp){
+                       tproxy_network="tcp,udp";
+                   } else if (GlobalConfig.inboundConfig.tproxy_use_tcp){
+                       tproxy_network="tcp";
+                   }else{
+                       tproxy_network="udp";
+                   }
+                   auto tproxyInSettings=GenerateDokodemoIN("",0,tproxy_network,10,true,0);
+                   tproxyInBoundObject.insert("settings", tproxyInSettings);
+
+                   QString sniffing_str=
```

Comment:
Please use c++ initializer list to initialize the QJsonObject, instead of using the QString and `JsonFromString` function.